apps/server/src/api-integration/integration.controller.ts

```diff
@@ -201,12 +201,11 @@ const actionHandlers: Record<string, ActionHandler> = {
       return { payload: 'success' };
     }
 
-    const timeToAdd = time * MILLIS_PER_SECOND; // frontend is seconds based
-    if (Math.abs(timeToAdd) > MILLIS_PER_HOUR) {
+    if (Math.abs(time) > MILLIS_PER_HOUR) {
```

Comment:
suggestion: this is outside of your scope, but this logic is overly complex and I wonder if we should cleanup

I would suggest something simpler like 
```ts
  addtime: (payload) => {
    const time = (() => {
      if (typeof payload === 'object' && payload !== null) {
        if ('add' in payload) return numberOrError(payload.add);
        if ('remove' in payload) return numberOrError(payload.remove) * -1;
      }
      return numberOrError(payload);
    })();

    if (Math.abs(time) > MILLIS_PER_HOUR) {
      throw new Error(`Payload too large: ${time}`);
    }

    if (time !== 0) {
      runtimeService.addTime(time);
    }

    return { payload: 'success' };
  },
```